lib/WorseReflection/Core/Util/NodeUtil.php

```diff
@@ -96,13 +96,16 @@ public static function nodeContainerClassLikeDeclaration(Node $node): ?Node
     }
 
     /**
-     * @param Token|Node|mixed $nodeOrToken
+     * @param Token|Node|ObjectCreationExpression|mixed $nodeOrToken
      */
     public static function nameFromTokenOrNode(Node $node, $nodeOrToken): string
     {
         if ($nodeOrToken instanceof Token) {
             return (string)$nodeOrToken->getText($node->getFileContents());
         }
+        if ($nodeOrToken instanceof ObjectCreationExpression) {
```

Comment:
this is method is explicitly about getting the `name` frm a Node or Token, it's not about annonymous classes. you can add a separate method or probably there is something better.